API.md

```diff
@@ -0,0 +1,1246 @@
+        -   `params.options.groupTraits` **[boolean][60]**  (optional, default `false`)
+    -   `params.userHandlerOptions` **[Object][57]?** deprecated
+
+**Examples**
+
+```javascript
+const { smartNotifierHandler } = require("hull/lib/utils");
+const app = express();
+
+const handler = smartNotifierHandler({
+  handlers: {
+    'ship:update': function(ctx, messages = []) {},
+    'segment:update': function(ctx, messages = []) {},
+    'segment:delete': function(ctx, messages = []) {},
+    'account:update': function(ctx, messages = []) {},
+    'user:update': function(ctx, messages = []) {
+      console.log('Event Handler here', ctx, messages);
+      // ctx: Context Object
+      // messages: [{
+      //   user: { id: '123', ... },
+      //   segments: [{}],
+      //   changes: {},
+      //   events: [{}, {}]
+      //   matchesFilter: true | false
+      // }]
+      // more about `smartNotifierResponse` below
+      ctx.smartNotifierResponse.setFlowControl({
+        type: 'next',
+        size: 100,
+        in: 5000
+      });
+      return Promise.resolve();
+    }
+  },
+  options: {
+    groupTraits: false
+  }
+});
+
+connector.setupApp(app);
+app.use('/notify', handler);
+```
+
+Returns **[Function][61]** expressjs router
+
+### superagentErrorPlugin
+
+This is a general error handling SuperAgent plugin.
+
+It changes default superagent retry strategy to rerun the query only on transient
+connectivity issues (`ECONNRESET`, `ETIMEDOUT`, `EADDRINFO`, `ESOCKETTIMEDOUT`, `ECONNABORTED`).
+So any of those errors will be retried according to retries option (defaults to 2).
+
+If the retry fails again due to any of those errors the SuperAgent Promise will
+be rejected with special error class TransientError to distinguish between logical errors
+and flaky connection issues.
+
+In case of any other request the plugin applies simple error handling strategy:
+every non 2xx or 3xx response is treated as an error and the promise will be rejected.
+Every connector ServiceClient should apply it's own error handling strategy by overriding `ok` handler.
+
+**Parameters**
+
+-   `options` **[Object][57]**  (optional, default `{}`)
+    -   `options.retries` **[Number][59]?** Number of retries
+    -   `options.timeout` **[Number][59]?** Timeout for request
+
+**Examples**
+
+```javascript
+superagent.get("http://test/test")
+  .use(superagentErrorPlugin())
+  .ok((res) => {
+    if (res.status === 401) {
+      throw new ConfigurationError();
+    }
+    if (res.status === 429) {
+      throw new RateLimitError();
+    }
+    return true;
+  })
+  .catch((error) => {
+    // error.constructor.name can be ConfigurationError, RateLimitError coming from `ok` handler above
+    // or TransientError coming from logic applied by `superagentErrorPlugin`
+  })
+```
+
+Returns **[Function][61]** function to use as superagent plugin
+
+### superagentInstrumentationPlugin
```

Comment:
Seems weird to have this and a `Infra.InstrumentationAgent` totally unrelated, with completely different usage. But that's probably a refactor.